ios/App/Shared/player/AudioPlayer.swift

```diff
@@ -378,32 +375,24 @@ class AudioPlayer: NSObject {
             
             setupQueueItemStatusObserver()
         } else {
-            NSLog("Seeking in current item \(to)")
+            NSLog("SEEK: Seeking in current item \(to)")
             let currentTrackStartOffset = playbackSession.audioTracks[self.currentTrackIndex].startOffset ?? 0.0
             let seekTime = to - currentTrackStartOffset
             
             self.audioPlayer.seek(to: CMTime(seconds: seekTime, preferredTimescale: 1000)) { [weak self] completed in
-                if !completed {
-                    NSLog("WARNING: seeking not completed (to \(seekTime)")
-                }
+                guard completed else { return NSLog("SEEK: WARNING: seeking not completed (to \(seekTime)") }
+                guard let self = self else { return }
                 
                 if continuePlaying {
-                    self?.play()
+                    self.resumePlayback()
```

Comment:
This change prevents a condition where seeking would occur within seeking, due to `play()` also needing to seek to the proper start time (even outside of `seekBack` conditions).